portfolio/src/main/webapp/script.js

```diff
@@ -42,36 +42,24 @@ function addRandomFact() {
 }
 
 
-function getRandomGreeting() {
-  console.log('Fetching a greeting.');
 
-  // The fetch() function returns a Promise because the request is asynchronous.
-  const responsePromise = fetch('/data');
-
-  // When the request is complete, pass the response into handleResponse().
-  responsePromise.then(handleResponse);
-}
-
-/**
- * Handles response by converting it to text and passing the result to
- * addQuoteToDom().
- */
-function handleResponse(response) {
-  console.log('Handling the response.');
-
-  // response.text() returns a Promise, because the response is a stream of
-  // content and not a simple variable.
-  const textPromise = response.text();
-
-  // When the response is converted to text, pass the result into the
-  // addQuoteToDom() function.
-  textPromise.then(addQuoteToDom);
+/** Fetches comments from the server and adds them to the DOM. */
+function loadComments() {
+  fetch('/data').then(response => response.json()).then((UserComments) => {
+    const commentListElement = document.getElementById('comment-list');
+    UserComments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+    })
+  });
 }
 
-/** Adds a random quote to the DOM. */
-function addQuoteToDom(quote) {
-  console.log('Adding quote to dom: ' + quote);
+/** Creates an element that represents a task, including its delete button. */
+function createCommentElement(task) {
+  const commentElement = document.createElement('li');
+  taskElement.className = 'comment';
 
-  const quoteContainer = document.getElementById('quote-container');
-  quoteContainer.innerText = quote;
-}
+  const commentElement = document.createElement('span');
```

Comment:
You're reusing the `commentElement` variable (another one on line 58) I believe.

Also, not sure if this is part of the walkthrough, but I'm not a fan of it using `const` in this situation.  Usually `const` means the variable will not be modified, but the appendChild calls below says otherwise.  I would instead use `let` or `var`.

Reply:
They used const in the examples/walkthroughs but I'll try it with let/var instead and see what happens!